src/main/java/org/opensearch/geospatial/ip2geo/action/RestPutDatasourceHandler.java

```diff
@@ -62,6 +65,9 @@ protected RestChannelConsumer prepareRequest(final RestRequest request, final No
         if (putDatasourceRequest.getUpdateInterval() == null) {
             putDatasourceRequest.setUpdateInterval(TimeValue.timeValueDays(clusterSettings.get(Ip2GeoSettings.DATASOURCE_UPDATE_INTERVAL)));
         }
+
+        // Call to validate if URL is in a deny-list or not.
+        urlDenyListChecker.toUrlIfNotInDenyList(putDatasourceRequest.getEndpoint());
```

Comment:
if we get an exception from this function what message is send to the customer? Can we catch all those exceptions and send a proper exception with readable message to customer